pkg/webhook/services/validating.go

```diff
@@ -93,6 +97,10 @@ func (r *handler) handleService(ctx context.Context, clt client.Client, decoder
 		}
 	}
 
+	if svc.Spec.Type == corev1.ServiceTypeNodePort && tnt.GetAnnotations()[enableNodePortsAnnotation] == "false" {
```

Comment:
Don't we have to check if `tnt.GetAnnotations()` is not nil here to avoid nil pointer dereference?

Reply:
i've got a test case for this - https://github.com/clastix/capsule/pull/259/files#diff-4e5ed4f1f8da5436910e7528461ba68f9ceb5eb373755b3cdb857856c7e36d1dR33-R44
and everything seems ok :)